features/313-alternative-to-auth-hooks/proposal.md

```diff
@@ -0,0 +1,108 @@
+# Alternative to Auth hooks
+
+## How Auth hooks look like now
+
+Every auth operation has its corresponding event. For example, `signup` operation has the `signup` event.
+
+For each event, the hook can be `async` and `sync`, `before` and `after`. Therefore the `signup` operation
+has the following hooks
+
+- `before_signup_sync`
+- `before_signup_async`
+- `after_signup_sync`
+- `after_signup_async`
+
+## Usage of auth hooks
+
+Suppose we are writing an application that needs to integrate with an existing user system.
+Let's name that external user system X.
+The app has its own user system which is clearly the Auth gear. During signup, the user of
```

Comment:
"which is clearly the Auth gear"?

Reply:
I mean the application is using skygear next so its auth things (signup, login, etc) are provided by the auth gear.